src/components/TodoList/TodoList.jsx

```diff
@@ -4,18 +4,13 @@ import TodoItem from "./TodoItem.jsx";
 const TodoList = ({ todos = [], isLoading }) => {
     const [ongoingTodos, setOngoingTodos] = useState([]);
     const [completedTodos, setCompletedTodos] = useState([]);
-
```

Comment:
Having 2 states and separating the todos could potentially cause some issues.
Here is a link to a video discussing kind of the issue (not exactly the same scenario): https://www.youtube.com/watch?v=tz0fDABt67g

It would be better if you have 1 state for `todos`
and have two regular variables `completedTodos` and `ongoingTodos`.

```
const completedTodos = todos.filter(x => x.completed)
const ongoingTodos = todos.filter(x => !x.completed)
```
That way, you only need to render once every time your state changes
React 18 does some state batching magic to try to eliminate multi renders, but you cannot guarantee when it will work and when it will not.